src/main/java/pl/onsight/wypozyczalnia/service/impl/CartServiceImpl.java

```diff
@@ -44,4 +54,41 @@ public void addDiscountToCart(Cart cart, double discount) {
     public void removeProductsFromCart(Cart cart) {
         cart.setProducts(new LinkedList<>());
     }
+
+    @Override
+    public boolean removeProductFromCart(Cart cart, ProductEntity product) {
+        List<ProductEntity> products = cart.getProducts();
+        if (products.contains(product)) {
+            products.remove(product);
+            cart.setProducts(products);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean removeProductsOneTypeFromCart(Cart cart, ProductEntity product) {
+        List<ProductEntity> products = cart.getProducts();
+        return products.removeIf(productEntity -> productEntity.equals(product));
+
+    }
+
```

Comment:
jak robisz returna w ifie to nie jest potrzebny else albo return mozna po prostu poza ifami zrobic na koncu